starknet_devnet/blueprints/rpc/structures/payloads.py

```diff
@@ -1006,14 +1006,48 @@ def rpc_map_traces(
     return result
 
 
+def gateway_to_rpc_invocation(invocation_dict: dict) -> Dict[str, Dict]:
+    """
+    Convert function invocation result from gateway to RPC.
+    """
+    if invocation_dict is None:
+        return invocation_dict
+
+    # All unnecessary properties should be removed except this mentioned here.
+    accepted_properties = {
+        "contract_address",
+        "entry_point_selector",
+        "calldata",
+        "caller_address",
+        "entry_point_type",
+        "class_hash",
+        "result",
+        "calls",
+        "events",
+        "messages",
+    }
+    keys = invocation_dict.copy().keys()
+    for key in keys:
+        if key not in accepted_properties:
+            del invocation_dict[key]
+
+    return invocation_dict
```

Comment:
```suggestion
    return { key: invocation_dict[key] for key in invocation_dict if key in accepted_properties }
```

Reply:
Fixed in 9770c8567a97df84d0208b0826854ad28dd4d2ff